flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/utils/LookupJoinUtil.java

```diff
@@ -58,4 +71,50 @@ public FieldRefLookupKey(int index) {
     private LookupJoinUtil() {
         // no instantiation
     }
+
+    /** Gets LookupFunction from temporal table according to the given lookup keys. */
+    public static UserDefinedFunction getLookupFunction(
+            RelOptTable temporalTable, Collection<Integer> lookupKeys) {
+
+        List<Integer> lookupKeyIndicesInOrder = new ArrayList<>(lookupKeys);
+        lookupKeyIndicesInOrder.sort(Integer::compareTo);
```

Comment:
nit: `CommonExecLookupJoin#getOrderedLookupKeys` can be moved into `LookupJoinUtil` and then this method can reuse it.